app/models/requests/scan.rb

```diff
@@ -21,13 +21,14 @@ def send_approval_status!
     RequestStatusMailer.request_status_for_scan(self).deliver_later if notification_email_address.present?
   end
 
+  # Returns true if a background job was enqueued.
   def submit!
-    SubmitIlliadRequestJob.perform_later(id)
-
-    # This ensures that only scan rules with a destination get sent to the ILS.
-    # We no longer want to send SAL3 requests to the ILS as this is handled by the ILLiad integration.
-    # SAL1/2 requests still go to the ILS at this time.
-    send_to_ils_later! if scan_destination&.dig(:patron_barcode).present?
+    SubmitIlliadRequestJob.perform_later(id).tap do
+      # This ensures that only scan rules with a destination get sent to the ILS.
+      # We no longer want to send SAL3 requests to the ILS as this is handled by the ILLiad integration.
+      # SAL1/2 requests still go to the ILS at this time.
+      send_to_ils_later! if scan_destination&.dig(:patron_barcode).present?
+    end
```

Comment:
was this bug introduced because something expects `submit!` to return whatever the output of `perform` methods is? i don't really understand the implications of this change.